src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs

```diff
@@ -12,34 +12,73 @@ namespace Microsoft.PowerShell.Commands
     [OutputType(typeof(System.IO.FileInfo))]
     public class NewTemporaryFileCommand : Cmdlet
     {
+        private string _extension = ".tmp";
```

Comment:
I disagree with using `Path.GetRandomFileName()` in the special case the Extension parameter not being present because this function does not only give you a random file name but already creates the file for you. Therefore it would highly increase the complexity of the code because then one ends up with 2 different ways of generating the file and having to handle the errors differently in 2 places. I strongly believe in having only one common code path. Can you please re-consider?